src/UriBuilder.Fluent/TerribleDevUriExtensions.cs

```diff
@@ -68,6 +68,65 @@ public static UriBuilder WithParameter(this UriBuilder bld, string key, IEnumera
         }
 
         /// <summary>
+        /// Appends a fragments parameter with a key, and many values. Multiple values will be comma seperated. If only 1 value is passed and its null or value, the key will be added to the fragment.
+        /// </summary>
+        /// <param name="bld"></param>
+        /// <param name="key"></param>
+        /// <param name="values"></param>
+        /// <returns></returns>
+        public static UriBuilder WithFragment(this UriBuilder bld, string key, params string[] values) => bld.WithFragment(key, valuesEnum: values);
+
+        /// <summary>
+        /// Appends fragments from dictionary
+        /// </summary>
+        /// <param name="bld"></param>
+        /// <param name="fragmentDictionary"></param>
+        /// <exception cref="ArgumentNullException"></exception>
+        /// <returns></returns>
+        public static UriBuilder WithFragment(this UriBuilder bld, IDictionary<string, string> fragmentDictionary)
+        {
+            if(fragmentDictionary == null) throw new ArgumentNullException(nameof(fragmentDictionary));
+            foreach(var item in fragmentDictionary)
+            {
+                bld.WithFragment(item.Key, item.Value);
+            }
+            return bld;
+        }
+
+        /// <summary>
+        /// Appends a fragments with a key, and many values. Multiple values will be comma seperated. If only 1 value is passed and its null or value, the key will be added to the fragment.
+        /// </summary>
+        /// <param name="bld"></param>
+        /// <param name="key"></param>
+        /// <param name="valuesEnum"></param>
+        /// <returns></returns>
+        public static UriBuilder WithFragment(this UriBuilder bld, string key, IEnumerable<object> valuesEnum)
+        {
+            if(string.IsNullOrWhiteSpace(key))
+            {
+                throw new ArgumentNullException(nameof(key));
+            }
+            if(valuesEnum == null)
+            {
+                valuesEnum = new string[0];
+            }
+            var intitialValue = string.IsNullOrWhiteSpace(bld.Fragment) ? "" : $"{bld.Fragment.TrimStart('?')}&";
+            var sb = new StringBuilder($"{intitialValue}{key}");
```

Comment:
I'm assuminthas a total copy from `WithParameter` do you mind refactoring this part out into a function and sharing that between the two? I assume there is a subset of stuff that could be abstracted out and re-used